src/tasks/fetch.ts

```diff
@@ -1,3 +1,39 @@
+import assign from "../util/assign";
+import prefixKeys from "../util/prefixKeys";
+
 import Task from "./task";
+import { asyncGetEntry, normalizeEntry } from "../lib/resourceTiming";
+
+class Fetch extends Task {
+  public constructor(config: Config, taskData: TaskData) {
+    super(config, taskData);
+  }
+
+  private fetchObjectAndId(): Promise<string> {
+    return fetch(this.data.resource).then(
+      (res): string => res.headers.get("X-Datacenter") || ""
+    );
+  }
+
+  protected test(): Promise<ResourceTimingEntry> {
+    let subjectId: string;
+
+    return Promise.all([
```

Comment:
What happens if this errors out? No `.catch` ?

Reply:
As all Tasks are consumed by the task scheduler in `src/worker.ts`; it is the responsibility of the scheduler to handle the execution errors at a global level rather than within the task. I.e. we intentionally want the error to bubble up to the caller.